git-fetch --jobs==0 if no configuration

gerhardol opened the pull request:
Fixes #9937

## Proposed changes

Override the Git default of one parallel job.

If the user has set fetch.parallel or submodule.fetchJobs the
the user config value is set (or git default if just one is set).

Config check adds 4ms the first check for me and <1ms the second (when the value is cached).

## Test methodology <!-- How did you ensure quality? -->

Manual
I guess this could be tested, someone is welcome to contribute.

## Merge strategy

I agree that the maintainer squash merge this PR (if the commit message is clear).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

mstv commented on GitCommands/Git/GitModule.cs:
```diff
@@ -1522,9 +1522,14 @@ private ArgumentString GetFetchArgs(string? remote, string? remoteBranch, string
             // TODO return ArgumentBuilder and add special case ArgumentBuilder.Add(ArgumentBuilder childBuilder)
             return new ArgumentBuilder
             {
+                { remoteJobs && string.IsNullOrWhiteSpace(branchArguments), "--multiple" },
```
The Git documentation mentions this option. Though it seems to be implicit, i.e. not necessary.

gerhardol replied:
Removed --multiple

mstv reviewed (commented):
LGTM
Although I would have chosen the more active approach: I would have run the regarding `git config` command if the value has not been configured yet.

gerhardol commented:
> LGTM Although I would have chosen the more active approach: I would have run the regarding `git config` command if the value has not been configured yet.

You mean runnig git-config? That would be a different way using config from the rest of GE and add some tenths (50?) ms for at least every repo.

Also fixed the tests and added tests for config options.
